yjit/src/options.rs

```diff
@@ -59,6 +62,7 @@ pub static mut OPTIONS: Options = Options {
     num_temp_regs: 5,
     gen_stats: false,
     gen_trace_exits: false,
+    trace_exits_sample_rate: 0,
```

Comment:
```suggestion
    trace_exits_sample_rate: 101,
```

Should probably make the default a prime number such as 101

Reply:
I think we always overwrite this value when it is used and don't use this default value. Maybe we should add a way to enable sampling without setting a sample rate? I am tempted keep it this way and always require the rate since that way users know what they are getting.

Reply:
Oh I see what you mean.

No I think the way you have it is right.

But... Could you add a check that the sample rate is a prime number? 😅 

Use the `O(sqrt(N))` method for simplicity: https://www.rookieslab.com/posts/fastest-way-to-check-if-a-number-is-prime-or-not

Reply:
Sure! This is a thing I don't know about sampling profilers, is it typical to only sample every prime number of samples?

Maybe this makes missing important samples due to instruction cycles less likely because the instruction cycle has to be exactly that number of instructions instead of a multiple of the sample rate?

Reply:
It's standard practice with hash maps to have the table size be a prime number to avoid artifacts. I'm assuming it could also help prevent issues here.

Reply:
IMO using a prime number (or just not an extremely "even" number) is a best practice, but enforcing that is overkill. When using a sampling profiler like ex. `perf` I just use `-F 99` (because I like everyone else copy from https://www.brendangregg.com/perf.html) and don't lose sleep over it.

Reply:
I updated this to be a warning instead of an error. Maybe that will helpfully warn people without preventing them from running their test. For example I might want a fairly sarge sample rate, but I don't want to find a large prime number.

Reply:
Fair enough. Thanks Adam :)